extract_unique_sensors_01.py

```diff
@@ -86,9 +86,9 @@ def extract_all_sensors(sensors_data, sensor_name):
     assert isinstance(sensor_name, str), "Expected a string."
     assert isinstance(sensors_data, list), "Expected a list."
     candidate_lines = []
-    [
-        candidate_lines.append(candidate_line)for sensor_data in sensors_data
-        if sensor_data.startswith(sensor_name) for candidate_line in sensor_data.split()
+    candidate_lines = [
+        candidate_line for sensor_data in sensors_data if sensor_data.startswith(sensor_name)
+        for candidate_line in sensor_data.split()
```

Comment:
We can chat about what you are attempting to do here, because I see there are 2 loops you are iterating through in the list comp and I don't understand what you are trying to accomplish.

Reply:
Okay, thanks.